packages/rest-hooks/src/resource/index.ts

```diff
@@ -1,6 +0,0 @@
-import Resource from './Resource';
-import SimpleResource from './SimpleResource';
```

Comment:
could someone be deep-importing this file?

Reply:
theoretically i guess. i don't see why they would with them being available at the top level tho.

Reply:
Interesting...I feel like before 'exports' in package.json this was inherently broken as main vs module was never really consistent between environments

Reply:
I'm curious to learn about the tree shaking limitations. Are there blog posts or something documenting this?

Reply:
Not that i know of. It’s just that tree-shaking is always best effort, and bundle sizes are always smaller when tree shaking isn’t needed. Coinbase no longer allows anything but deep imports in most of its internal code for this reason.

Reply:
If this is in web, seems like lack of webpack 5 is the real problem as webpack 5 does dead code elimination as well.

Reply:
All treeshaking, regardless of bundler, can’t ever possibly do as good a job as not needing it in the first place ¯\\\_(ツ)\_/¯ 

manifest exports/barrel exports/whatever they’re called are an antipattern, albeit a common one, and it’s best avoided.

Reply:
You mean like `import * as React from 'react';`? That's very different than `import { useMemo } from 'react';`

Reply:
The first one breaks everything, the second one not only doesn't break tooling, it allows minifiers to do wonders on the imported pieces

Reply:
Right - but unless `react` is designed quite carefully, `import { useMemo }` still needs to *evaluate* all that code because of side effects and whatnot.